ui/desktop/src/config.ts

```diff
@@ -10,10 +9,68 @@ export const getSecretKey = (): string => {
   return window.appConfig.get('secretKey');
 }
 
+// Environment variables type matching Rust's Envs
+export type Envs = Record<string, string>;
+
+// SystemConfig type matching the Rust version
+export type SystemConfig =
+  | {
+      type: "sse";
+      uri: string;
+      envs?: Envs;
+    }
+  | {
+      type: "stdio";
+      cmd: string;
+      args: string[];
+      envs?: Envs;
+    }
+  | {
+      type: "builtin";
+      name: string;
+    };
+
+// Extend Goosed with a new system configuration
+export const extendGoosed = async (config: SystemConfig) => {
+  // allowlist the CMD for stdio type
+  if (config.type === "stdio") {
+    const allowedCMDs = ['npx', 'uvx', 'goosed'];
+    if (!allowedCMDs.includes(config.cmd)) {
+      console.error(`System ${config.cmd} is not supported right now`);
+      return;
+    }
+
+    // if its goosed - we will update the path to the binary
+    if (config.cmd === 'goosed') {
+      config = {
+        ...config,
+        cmd: await window.electron.getBinaryPath('goosed')
+      };
+    }
+  }
+
+  try {
+    const response = await fetch(getApiUrl('/systems/add'), {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Secret-Key': getSecretKey(),
+      },
+      body: JSON.stringify(config)
+    });
+
+    if (!response.ok) {
+      throw new Error(`Failed to add system config: ${response.statusText}`);
+    }
+    console.log(`Successfully added system config: ${JSON.stringify(config)}`);
+  } catch (error) {
+    console.log(`Error adding system config:`, error);
+  }
+};
 
-// add MCP system from a goose://extension url 
-export const addMCPSystem = async (url: string) => {
-  console.log("adding MCP from URL", url);
+// Extend Goosed from a goose://extension URL
+export const extendGoosedFromUrl = async (url: string) => {
```

Comment:
maybe should be `openFromGoosedLink` ?

Reply:
this doesn't make a new window though right? just adds a new extension to the existing agent in place?

Reply:
true - yes you are right